spinalcordtoolbox/reports/qc2.py

```diff
@@ -316,21 +316,18 @@ def sct_deepseg(
         fig.set_size_inches(*size_fig, forward=True)
         FigureCanvas(fig)
         ax = fig.add_axes((0, 0, 1, 1))
-        colormaps = {
-            # Single color (Red, Cyan)
-            'seg': ["#ff0000", "#00ffff"],
-            # Two-tone colormap (Red-darkred, Cyan-darkcyan)
-            'softseg': [color.ListedColormap(["#4d0000", "#ff0000"]),
-                        color.ListedColormap(["#004d4d", "#00ffff"])],
-        }
+        colormaps = [color.ListedColormap(["#ff0000"]),  # Red for first image
+                     color.ListedColormap(["#00ffff"])]  # Cyan for second
         for i, image in enumerate([img_seg_sc, img_seg_lesion]):
             if not image:
                 continue
             img = mosaic(image, centers)
             img = np.ma.masked_equal(img, 0)
             ax.imshow(img,
-                      cmap=(colormaps['seg'] if check_image_kind(image.data) == 'seg' else colormaps['softseg'])[i],
+                      cmap=colormaps[i],
                       norm=color.Normalize(vmin=0.5, vmax=1),
+                      # img==1 -> opaque, but soft regions -> more transparent as value decreases
+                      alpha=(img / img.max()),  # scale to [0, 1]
```

Comment:
If we really want to make sure we've scaled to [0, 1], maybe line 325 should mask out negative values as well (even though we don't expect them):
```py
img = np.ma.masked_less_equal(img, 0)
img.set_fill_value(0)
```
(I'm not sure what impact the `fill_value` has, but `masked_equal` sets it differently from `masked_less_equal`.)

Reply:
> (I'm not sure what impact the `fill_value` has, but `masked_equal` sets it differently from `masked_less_equal`.)

Wow, good catch!

```python
>>> img.fill_value
1e+20
```

That said, the `set_fill_value` doesn't seem to change the QC report itself? So, I too am left wondering what impact `fill_value` has, and it seems like we're not alone:

- https://github.com/numpy/numpy/issues/20850

I added my 2 cents in a follow-up comment, since it seems like multiple folks are getting tripped up by this behavior!!